daemon/k8s_watcher.go

```diff
@@ -317,19 +317,24 @@ func k8sErrorHandler(e error) {
 
 // blockWaitGroupToSyncResources ensures that anything which waits on waitGroup
 // waits until all objects of the specified resource stored in Kubernetes are
-// received by informer. Fatally exits if syncing these initial objects fails.
-func blockWaitGroupToSyncResources(waitGroup *sync.WaitGroup, informer cache.Controller, resourceName string) {
+// received by the informer and processed by the given Function Queue.
+// Fatally exits if syncing these initial objects fails.
+func blockWaitGroupToSyncResources(waitGroup *sync.WaitGroup, informer cache.Controller,
+	resourceName string, queue *serializer.FunctionQueue) {
+
 	waitGroup.Add(1)
 	go func() {
-		completed := make(<-chan struct{})
 		scopedLog := log.WithField("kubernetesResource", resourceName)
 		scopedLog.Debug("waiting for cache to synchronize")
-		if ok := cache.WaitForCacheSync(completed, informer.HasSynced); !ok {
+		if ok := cache.WaitForCacheSync(wait.NeverStop, informer.HasSynced); !ok {
 			// Fatally exit it resource fails to sync
 			scopedLog.Fatalf("failed to wait for cache to sync")
 		}
-		scopedLog.Debug("cache synced")
-		waitGroup.Done()
+		queue.Enqueue(func() error {
```

Comment:
Add a comment indicating why we need to add this to the function queue. The reasoning is subtle, in that while the informer indicates that it has "synced", we want to make sure that Cilium has properly consumed said event, and only then can we indicate that the information has been processed by Cilium.

Reply:
Done